src/attributes/classes.js

```diff
@@ -14,7 +14,23 @@ function getClass( elem ) {
 
 jQuery.fn.extend( {
 	addClass: function( value ) {
-		var classes, elem, cur, curValue, clazz, j, finalValue,
+		var classes = [];
+		var proceed = false;
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnothtmlwhite ) || [];
+			proceed = true;
+		} else if ( value.constructor === Array ) {
+			for ( var count = 0; count < value.length; count++ ) {
+				if ( typeof value[ count ] === "string" && value[ count ] ) {
```

Comment:
This is not necessary; we don't want to support `.addClass(['a', 'b c'])`; if you pass an array you'll need to have every class as an individual entry in the array.